cst438_gradebook/src/main/java/com/cst438/controllers/GradeBookController.java

```diff
@@ -217,36 +159,95 @@ public void updateGradebook (@RequestBody GradebookDTO gradebook, @PathVariable(
 		
 	}
 	
-	// As an instructor, I can change the name of the assignment for my course.
-	@PutMapping("/gradebook/edit-name/{id}")
-	public void editAssignmentName (@RequestBody AssignmentDTO assignments, @PathVariable("id") Integer assignmentId ) {
-		
-		String email = "dwisneski@csumb.edu";  // user name (should be instructor's email) 
-		checkAssignment(assignmentId, email);  // check that user name matches instructor email of the course.
+	// As an instructor for a course , I can add a new assignment for my course.  
+	// The assignment has a name and a due date.
+	@PostMapping("/gradebook/add")
+	@Transactional
+	public AssignmentDTO addAssignment(@RequestBody AssignmentDTO assignmentDTO) {
+		Assignment duplicateAssignment = assignmentRepository.findById(assignmentDTO.assignmentId);
+			
+		if(duplicateAssignment == null) {
+			Assignment assignment = new Assignment();
+			assignment.setName(assignmentDTO.assignmentName);
+				
+			SimpleDateFormat formatter = new SimpleDateFormat("dd-MMM-yyyy", Locale.ENGLISH);
+			Date dueDate = null;
+			try {
+				dueDate = (Date) formatter.parse(assignmentDTO.dueDate);
+			} catch (ParseException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}	
+			assignment.setDueDate(dueDate);
+				
+			Course course = courseRepository.findByCourse_id(assignmentDTO.courseId);
+			assignment.setCourse(course);
+				
+			//assignment.setNeedsGrading(1);
+			Assignment newAssignment = assignmentRepository.save(assignment);
+				
+			return createAssignmentDTO(newAssignment);
+		} else {
+			throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Assignment already exists.");
+		}
+			
+	}
+	
 		
+	//As an instructor, I can change the name of the assignment for my course.
+		@PutMapping("/gradebook/edit-name/{id}")
+		@Transactional
+		public void editAssignmentName(@RequestBody AssignmentListDTO.AssignmentDTO assignments, @PathVariable("id") Integer assignmentId ) {
```

Comment:
Requiring the assignment id in both the POST body and the path variable is redundant but ultimately works.